Dockerfile

```diff
@@ -15,10 +14,15 @@
 
 COPY --from=builder /app .
 
-RUN pip install --upgrade pip && \
-    pip install -r requirements.txt
+RUN apt-get update && apt-get install -y \
+    libmariadb-dev \
+    && pip install --upgrade pip \
+    && pip install mysqlclient \
+    && pip install -r requirements.txt
 
 EXPOSE 8080
 
-# Run database migrations and start the Django application
-ENTRYPOINT ["sh", "-c", "python manage.py migrate && python manage.py runserver 0.0.0.0:8080"]
+COPY wait-for-it.sh /app/wait-for-it.sh
+RUN chmod +x /app/wait-for-it.sh
+
+ENTRYPOINT ["sh", "-c", "./wait-for-it.sh mysql:3306 -- python manage.py migrate && python manage.py runserver 0.0.0.0:8080"]
```

Comment:
The `ENTRYPOINT` command uses `sh -c`, which can be problematic if you need to pass additional arguments to the container at runtime. Consider using an array syntax without `sh -c` if you need more flexibility.